scene/main/node.cpp

```diff
@@ -1718,6 +1718,20 @@ void Node::remove_child(Node *p_child) {
 	}
 }
 
+void Node::clear_children(bool p_deferred, bool p_include_internal) {
+	if (p_deferred) {
+		for (int i = 0; i < get_child_count(p_include_internal); i++) {
```

Comment:
For even more performance, you could iterate on `data.children` directly, instead of using the cache.

Reply:
This means we're gonna have to add maintenance cost to it, oh well.

Reply:
Nvm i saw it wrong and i was trying to optimize too much.